src/NUnitFramework/framework/ExceptionExtensions.cs

```diff
@@ -0,0 +1,102 @@
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections;
+
+namespace NUnit.Framework
+{
+    internal static class ExceptionExtensions
+    {
+        /// <summary>
+        /// If <see cref="Exception.StackTrace"/> throws, returns "SomeException was thrown by the Exception.StackTrace
+        /// property." See also <see cref="Assert.GetEnvironmentStackTraceWithoutThrowing"/>.
+        /// </summary>
+#if !(NET35 || NETSTANDARD1_4)
+        // https://github.com/dotnet/coreclr/issues/19698 is also currently present in .NET Framework 4.7 and 4.8. A
+        // race condition between threads reading the same PDB file to obtain file and line info for a stack trace
+        // results in AccessViolationException when the stack trace is accessed even indirectly e.g. Exception.ToString.
+        [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions]
+#endif
+        public static string GetStackTraceWithoutThrowing(this Exception exception)
+        {
+            if (exception is null) throw new ArgumentNullException(nameof(exception));
+
+            try
+            {
+                return exception.StackTrace;
+            }
+            catch (Exception ex)
+            {
+                return ex.GetType().Name + " was thrown by the Exception.StackTrace property.";
+            }
+        }
+
+        /// <summary>
+        /// If <see cref="Exception.Message"/> throws, returns "SomeException was thrown by the Exception.Message
+        /// property."
+        /// </summary>
+#if !(NET35 || NETSTANDARD1_4)
+        // https://github.com/dotnet/coreclr/issues/19698 is also currently present in .NET Framework 4.7 and 4.8. A
+        // race condition between threads reading the same PDB file to obtain file and line info for a stack trace
+        // results in AccessViolationException when the stack trace is accessed even indirectly e.g. Exception.ToString.
+        [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions]
+#endif
+        public static string GetMessageWithoutThrowing(this Exception exception)
```

Comment:
My idea was that by making these extension methods, we would notice them in intellisense and choose them over the properties in future code.

Reply:
(Meant to add this comment earlier, but I missed that there's no 'Add single comment' button in the mobile verson, so it was a pending review comment. Lame!)